src/service/registry/registry.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/goharbor/harbor/src/pkg/reverseproxy"
+	"github.com/goharbor/harbor/src/service/registry/handler"
+	"github.com/goharbor/harbor/src/service/registry/middleware"
+	"github.com/gorilla/mux"
```

Comment:
I feel gorilla/mux may be a little heavy for this task, did you investigate other options?

https://github.com/julienschmidt/go-http-routing-benchmark

Reply:
Why it is heavy? It's powerful and used commonly, it's easy for us to debug if get any error. And we can set middlewares for different routers easily